core/workspace_svg.ts

```diff
@@ -2143,8 +2142,8 @@ export class WorkspaceSvg extends Workspace implements IASTNodeLocationSvg {
    *
    * @param comment comment to add.
    */
-  override addTopComment(comment: OldWorkspaceComment) {
-    this.addTopBoundedElement(comment as OldWorkspaceCommentSvg);
+  override addTopComment(comment: WorkspaceComment) {
```

Comment:
Are these just not being updated to RenderedWorkspaceComment for backwards compatibility?

Reply:
That would break type compatibility :/ You can't have a subclass that expects a more restricted type than the superclass does. We never actually pass the base class to this function, so from a functional perspective we should be able to do that. But typescript doesn't know that so it prevents us from defining the types that way.

This is all knock on effects of the way we've set up rendered vs headless workspaces :/

Reply:
Oh yeah, missed the `override`, that makes sense. Thanks!